src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/FixCaseApiService.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ecm.common.model.ccd.CCDRequest;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.multiples.SubmitMultipleEvent;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.MultiplesHelper;
+import uk.gov.hmcts.ethos.replacement.docmosis.service.excel.MultipleCasesReadingService;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service("fixCaseApiService")
+public class FixCaseApiService {
+
+    private final MultipleCasesReadingService multipleCasesReadingService;
+
+    @Value("${ccd_gateway_base_url}")
+    private String ccdGatewayBaseUrl;
+
+    public void checkUpdateMultipleReference(CCDRequest ccdRequest, String userToken) {
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        if (caseData.getEcmCaseType().equals("Multiple")
+                && (caseData.getMultipleReference().length() == 16
+                || caseData.getMultipleReferenceLinkMarkUp() == null)) {
+            String multipleCaseTypeId = ccdRequest.getCaseDetails().getCaseTypeId() + "_Multiple";
+            List<SubmitMultipleEvent> submitMultipleEvents =
+                    retrieveMultiCase(userToken, multipleCaseTypeId, caseData.getMultipleReference());
```

Comment:
Based on the code in ecm-common this is going to be wrong - caseData.getMultipleReference() isn't going to be the 16 digit CCD number. You need caseDetails.caseId